src/inline-assembly.md

```diff
@@ -477,3 +478,143 @@ To avoid undefined behavior, these rules must be followed when using function-sc
+
+### Directives Support
+
+Inline assembly supports a subset of the directives supported by both GNU AS and LLVM's internal assembler, given as follows.
+The result of using other directives is assembler-specific (and may cause an error, or may be accepted as-is).
+
+If inline assembly includes any "stateful" directive that modifies how subsequent assembly is processed, the block must undo the effects of any such directives before the inline assembly ends.
+
+The following directives are guaranteed to be supported by the assembler:
+
+- `.2byte`
+- `.4byte`
+- `.8byte`
+- `.align`
+- `.ascii`
+- `.asciz`
+- `.balign`
+- `.balignl`
+- `.balignw`
+- `.balign`
+- `.balignl`
+- `.balignw`
+- `.bss`
+- `.byte`
+- `.comm`
+- `.data`
+- `.def`
+- `.double`
+- `.endef`
+- `.equ`
+- `.equiv`
+- `.eqv`
+- `.fill`
+- `.float`
+- `.lcomm`
+- `.inst`
+- `.long`
+- `.octa`
+- `.option`
+- `.p2align`
+- `.pushsection`
+- `.popsection`
+- `.quad`
+- `.scl`
+- `.section`
+- `.set`
+- `.short`
+- `.skip`
+- `.sleb128`
+- `.space`
+- `.string`
+- `.text`
+- `.uleb128`
+- `.word`
+
+The following directives are guaranteed to be supported for `global_asm` only:
+
+- `.alt_entry`
+- `.globl`
+- `.global`
+- `.private_extern`
+- `.size`
+- `.type`
+
+
+#### Target Specific Directive Support
+
+##### Dwarf Unwinding
+
+The following directives are supported on ELF targets that support DWARF unwind info:
+
+
+- `.cfi_adjust_cfa_offset`
+- `.cfi_def_cfa`
+- `.cfi_def_cfa_offset`
+- `.cfi_def_cfa_register`
+- `.cfi_endproc`
+- `.cfi_escape`
+- `.cfi_lsda`
+- `.cfi_offset`
+- `.cfi_personality`
+- `.cfi_register`
+- `.cfi_rel_offset`
+- `.cfi_remember_state`
+- `.cfi_restore`
+- `.cfi_restore_state`
+- `.cfi_return_column`
+- `.cfi_same_value`
+- `.cfi_sections`
+- `.cfi_signal_frame`
+- `.cfi_startproc`
+- `.cfi_undefined`
+- `.cfi_window_save`
+
+
+##### Structured Exception Handling
+
+On targets with structured exception Handling, the following additional directives are guaranteed to be supported:
+
+- `.seh_endproc`
+- `.seh_endprologue`
+- `.seh_proc`
+- `.seh_pushreg`
+- `.seh_savereg`
+- `.seh_setframe`
+- `.seh_stackalloc`
+
+
+##### x86 (32-bit and 64-bit)
+
+On x86 targets, both 32-bit and 64-bit, the following additional directives are guaranteed to be supported:
+- `.att_syntax`
+- `.intel_syntax`
+- `.nops`
+
+Use of the `.att_syntax` and `.intel_syntax` with no parameters is supported, but the syntax must be restored to the option at entry (`.intel_syntax` without the `att_syntax` asm option, or `.att_syntax` with that option) or the behaviour is undefined (the output of the compiler may be corrupted as a result). Use of `.att_syntax` and `.intel_syntax` with an option (such as `.intel_syntax prefix` or `.att_syntax noprefix`) is unsupported and results in assembler-dependant behaviour. If operand interpolations are used between setting the syntax mode with one of these directives, and restoring it to the block's default, the behaviour is undefined.
```

Comment:
Minor tweaks, notably allowing parameters equivalent to the default and only disallowing non-default parameters.

```suggestion
Use of the `.att_syntax` and `.intel_syntax` directives with no parameters (or with parameters equivalent to the defaults) is supported, but the syntax must be restored to the option at entry (`.intel_syntax` without the `att_syntax` asm option, or `.att_syntax` with that option) or the behavior is undefined (the compiled output may be corrupted as a result). Use of `.att_syntax` and `.intel_syntax` with a non-default option (such as `.intel_syntax prefix` or `.att_syntax noprefix`) is unsupported. If operand interpolations are used between setting the syntax mode with one of these directives, and restoring it to the block's default, the behaviour is undefined.
```

Reply:
It may be nice to say an implementation just needs to support `.intel_syntax`/`.att_syntax` and nothing after that. This isn't a huge deal, though, it's just a note.

Reply:
In practice, handling a single fixed optional parameter that can only take a single value or be omitted entirely, and which has no semantic effect, doesn't seem like it'd be difficult.

I can easily imagine a compiler detecting them and complaining, but if it has the capability to do so, it should detect the whole directive instead and lint towards `options(...)`. I don't think the argument in particular makes that any harder.